src/lib.rs

```diff
@@ -80,30 +80,34 @@ fn split_hash(hash: &str) -> BcryptResult<HashParts> {
     let mut parts = HashParts {
         cost: 0,
         salt: "".to_owned(),
-        hash: "".to_owned()
+        hash: "".to_owned(),
     };
 
     for (i, part) in hash.split('$').enumerate() {
         match i {
             0 => (),
-            1 => match part {
-                "2y" | "2b" | "2a" => (),
-                _ => { return Err(BcryptError::InvalidPrefix); }
-            },
+            1 => {
+                match part {
+                    "2y" | "2b" | "2a" => (),
+                    supplied => {
+                        return Err(BcryptError::InvalidPrefix(supplied.to_string()));
```

Comment:
If this is something we want to do, I'm not sure this is the best approach to take.